wpilibc/src/main/native/cpp/apriltag/AprilTagFieldLayout.cpp

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#include "frc/apriltag/AprilTagFieldLayout.h"
+
+#include <algorithm>
+#include <system_error>
+
+#include <units/angle.h>
+#include <units/length.h>
+#include <wpi/json.h>
+#include <wpi/raw_istream.h>
+#include <wpi/raw_ostream.h>
+
+using namespace frc;
+
+AprilTagFieldLayout::AprilTagFieldLayout(std::string_view path) {
+  std::error_code error_code;
+
+  wpi::raw_fd_istream input{path, error_code};
+  if (error_code) {
+    throw std::runtime_error(fmt::format("Cannot open file: {}", path));
+  }
+
+  wpi::json json;
+  input >> json;
+
+  m_apriltags = json.get<std::vector<AprilTag>>();
+}
+
+AprilTagFieldLayout::AprilTagFieldLayout(const std::vector<AprilTag>& apriltags)
+    : m_apriltags(apriltags) {}
```

Comment:
I checked the code out locally, and clang-tidy said to pass by value and move.
```suggestion
AprilTagFieldLayout::AprilTagFieldLayout(std::vector<AprilTag> apriltags)
    : m_apriltags(std::move(apriltags)) {}
```